cookbook/networking/fetch-data.md

```diff
@@ -92,15 +92,23 @@ Now, we'll update the `fetchPost` function to return a `Future<Post>`. To do so,
 we'll need to:
 
   1. Convert the response body into a json `Map` with the `dart:convert` package
-  2. Convert the json `Map` into a `Post` using the `fromJson` factory.
+  2. If the server returns an "OK" response with a status code of 200, convert 
+  the json `Map` into a `Post` using the `fromJson` factory.
+  3. If the server returns an unexpected response, throw an error
 
 <!-- skip -->
 ```dart
 Future<Post> fetchPost() async {
-  final response = await http.get('https://jsonplaceholder.typicode.com/posts/1');
-  final responseJson = json.decode(response.body); 
-  
-  return Post.fromJson(responseJson); 
+  final response =
+      await http.get('https://jsonplaceholder.typicode.com/posts/1');
+
+  if (response.statusCode == 200) {
```

Comment:
Yep, I think that might be just a *bit* out of scope (I don't want to get too crazy on an intro recipe, and had initially even left this part out to keep it as simple as possible).

IMO, the Parsing logic should be tested separately. You can see an example of the client-side implementation here: https://github.com/brianegan/giphy_client

You can also setup "Contract Tests", which are committed to the back-end project's repository. If the backend removes something from the JSON payload, or delivers an unexpected result, it would fail the tests for the backend during a CI check. We had a lot of success with this at SoundCloud.

On the client-side: If the parsing fails due to an error, the Future will complete with an error and you can show an "Error Screen." You should report the parsing error to an error-reporting service, such as Sentry or Crashlytics. Then, you can see what's failing, write a test for the failure, then fix the parsing logic and ensure the test passes. 

Even better than all of this: Using something like protocol buffers,  which will generate the Code for Client and Server based on a "Gold Standard" spec.

So yah, it's a bit complex and kind of a whole topic on it's own, and I kinda worry it's a bit too much for this simple recipe. What are your thoughts?

Reply:
Yeah, that definitely sounds more like an article unto itself than something you could easily drop in here with a few sentences. Comment retracted.